### PR TITLE
Update asyncfile.nim

### PR DESCRIPTION
**Pull request by lihf8515**
AsyncFile add field filename, open getFileSize

### CODE REVIEW

File: lib/pure/asyncfile.nim [hunk 1]
```diff
@@ -68,14 +69,9 @@ else:
       result = O_RDWR
     result = result or O_NONBLOCK
 
-proc getFileSize(f: AsyncFile): int64 =
+proc getFileSize*(f: AsyncFile): int64 =
   ## Retrieves the specified file's size.
-  when defined(windows) or defined(nimdoc):
```
Comment (Araq) [inline lib/pure/asyncfile.nim]: Why remove the specialized Windows implementation? It should be faster.

**Comment by dom96** [comment]
Why don't you just save your files filename and use ``getFileSize`` yourself?

**Comment by Araq** [comment]
@dom96 should we close the PR?

**Comment by dom96** [comment]
Yes.

**Comment by lihf8515** [comment]
So remove the portable detection, because the reference to the GetFileSize module in the OS function, and this function has been done to deal with, and it needs a file name as a parameter.

**Pull request closed without merging**